config/hcp.go

```diff
@@ -95,6 +95,9 @@ type hcpConfig struct {
 
 	// profile is the user's organization id and project id
 	profile *profile.UserProfile
+
+	// noBrowserLogin is an option to not automatically open browser login in no valid auth method is found.
+	noBrowserLogin bool
```

Comment:
Rather than adding this boolean field, we could instead clear the contents of `oauth2Config`, which drives the whole browser login flow.